cmd/generate/main.go

```diff
@@ -19,6 +19,8 @@ func main() {
 	var (
 		sourceCodePath = flag.String("src-path", "",
 			"The path of source code your want to generate support from, it can be either a file or a directory. (required)")
+		directiveMapName  = flag.String("directive-map", "", `Name of the generated map variable.(required)`)
+		matchFuncName     = flag.String("match-func", "", `Name of the generated matchFunc.(required)`)
```

Comment:
In the other MR you had mentioned wanting to provide a helpful default here. I like that idea but thought having a wrapper script enforce a variable was provided to be really weird.

If this is can reasonably have a default, and it's just that lua/headers need to provide a value then have those provide the value -- that's exactly what optional args are for.

What was your thinking for switching to required?

Reply:
Originally I made `directive-map` and `match-func` optional just for convenience in this usercase: an user has a small dynamic module and he wants to use crossplane to parse configs contain that module(but we don't want him open a MR to add `//go:generate` for that module). In this case he can run `./cmd/generate/` directly, and pass the generated `MatchFunc` to `Parseoptions.DirectiveSources` and I thought providing default values for these two params save the user's time.

After you mentioned the problem, I agreed it's quite weird for maintainers to understand why we defined it as required in shell script but optional in `./cmd/generate`. Moreover, the default values not that make sense. `directive-map` is the name of the generated map, and `match-func` is the name of the generated matchFunc. Consider the former usercase, the user is generating a support file for a dynamic module(say it `module_a`), and the `directive-map` and `match-func` should be something relevant to `module_a`(like `aDirectives`, `MatchADirectives`). The union default values I provided before don't really make sense, and I'm also concerned `optional` may cause users to overlook them.

Reply:
```suggestion
		directiveMapName  = flag.String("directive-map", "", `Name of the generated map variable.(required)`)
		matchFuncName     = flag.String("match-func", "", `Name of the generated matchFunc. (required)`)
```